controllers/halayerset_controller_test.go

```diff
@@ -175,52 +178,104 @@ var _ = Describe("High Availability Layer Set CR", func() {
 })
 
 func verifyCmdCommands(action userAction) {
+	time.Sleep(time.Second * 2)
 	switch action {
 	case doNothing:
 		Eventually(func() int { return len(execCmdCommands) }, time.Second*10, time.Millisecond*10).Should(BeEquivalentTo(0))
 	case createCR:
-		Eventually(func() int { return len(execCmdCommands) }, time.Second*10, time.Millisecond*10).Should(BeEquivalentTo(1))
 		expectedCommands := [][]string{
-			{"pcs", "stonith", "create", originalFenceAgentName, fenceAgentType},
+			{"pcs", "stonith", "create", firstOrgFenceAgent, fenceAgentType},
+		}
+		forbiddenCommands := [][]string{
+			{"pcs", "resource", "remove"},
+			{"pcs", "stonith", "create", updatedFenceAgentName, fenceAgentType},
 		}
 		verifyExpectedCommands(expectedCommands)
+		verifyNoUnExpectedCommand(forbiddenCommands)
 	case deleteCR:
-		Eventually(func() int { return len(execCmdCommands) }, time.Second*10, time.Millisecond*10).Should(BeEquivalentTo(3))
 		expectedCommands := [][]string{
 			{"pcs", "status", "xml"},
-			{"pcs", "stonith", "create", originalFenceAgentName, fenceAgentType},
-			{"pcs", "resource", "remove", originalFenceAgentName, "--force"},
+			{"pcs", "stonith", "create", firstOrgFenceAgent, fenceAgentType},
+			{"pcs", "resource", "remove", firstOrgFenceAgent, "--force"},
+		}
+		forbiddenCommands := [][]string{
+			{"pcs", "resource", "remove", updatedFenceAgentName, "--force"},
+			{"pcs", "stonith", "create", updatedFenceAgentName, fenceAgentType},
 		}
 		verifyExpectedCommands(expectedCommands)
+		verifyNoUnExpectedCommand(forbiddenCommands)
 	case updateCR:
-		Eventually(func() int { return len(execCmdCommands) }, time.Second*10, time.Millisecond*10).Should(BeEquivalentTo(5))
 		expectedCommands := [][]string{
-			{"pcs", "stonith", "create", originalFenceAgentName, fenceAgentType},
+			{"pcs", "stonith", "create", firstOrgFenceAgent, fenceAgentType},
+			{"pcs", "stonith", "create", secondOrgFenceAgent, fenceAgentType},
 			{"pcs", "status", "xml"}, //appears twice
-			{"pcs", "resource", "remove", originalFenceAgentName, "--force"},
+			{"pcs", "resource", "remove", firstOrgFenceAgent, "--force"},
 			{"pcs", "stonith", "create", updatedFenceAgentName, fenceAgentType},
+			{"pcs", "stonith", "update", secondOrgFenceAgent},
 		}
+		forbiddenCommands := [][]string{
+			{"pcs", "resource", "remove", updatedFenceAgentName, "--force"},
+			{"pcs", "resource", "remove", secondOrgFenceAgent, "--force"},
+		}
+		//debug()
 		verifyExpectedCommands(expectedCommands)
+		verifyNoUnExpectedCommand(forbiddenCommands)
 	}
 
 }
 
+func debug() {
+	for i := 0; i < 30; i++ {
```

Comment:
Instead of this loop, why not right `time.Sleep(30 * time.Second)`?

Reply:
I've used it for debugging this code should be deleted.
Good catch !